src/admin/user/update-user.dto.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsOptional, IsString, Matches } from 'class-validator';
+import { IsBoolean, IsEmail } from 'class-validator';
+
+export class UpdateUserDto {
+  @ApiProperty()
+  @IsString()
+  firstName: string;
+
+  @ApiProperty()
+  @IsString()
+  lastName: string;
+
+  @ApiProperty()
+  @IsString()
+  displayName: string;
+
+  @ApiProperty()
+  @IsEmail()
+  email: string;
+
+  @ApiProperty()
+  @IsString()
+  @IsOptional()
```

Comment:
I think all columns must be optional. In case we want to change just one value.

Reply:
There is no need for that - we can "force" consumers to send all data and if something is changed it will be in that request, otherwise original content. This is how regular PUT works... what you said is correct for PATCH